src/server.rs

```diff
@@ -560,27 +515,32 @@ fn handle_put_property(
 
     let args = match body.as_object() {
         Some(args) => args,
-        None => return HttpResponse::BadRequest().finish(),
+        None => {
+            return HttpResponse::BadRequest().json(bad_request(
+                "Parsing request failed",
+                Some(body.into_inner()),
+            ))
+        }
     };
 
-    if !args.contains_key(property_name) {
-        return HttpResponse::BadRequest().finish();
-    }
+    let arg = if let Some(arg) = args.get(property_name) {
+        arg
+    } else {
+        return HttpResponse::BadRequest().json(bad_request(
+            "Request does not contain property key",
+            Some(json!(args)),
+        ));
+    };
```

Comment:
I think this is more concise. Unfortunately, I had no chance to try it out. What do you think?

```suggestion
    let arg = args.get(property_name).ok_or_else(|| HttpResponse::BadRequest().json(bad_request(
        "Request does not contain property key",
        Some(json!(args)),
    )))?;
```

Reply:
I agree, but the suggestion doesn't work.
Let's open another PR for this.